modules/standard/FFTW.chpl

```diff
@@ -222,29 +427,142 @@ module FFTW {
+
+  /* Specify that FFTW should try and find an optimized plan by
+     computing several FFTs and measuring their execution time.
+     This can consume some time.
+  */
+  // TODO: If/when we support defaults, might say something like: This
+  // is the default planning option.
+  extern const FFTW_MEASURE : c_uint;
+
+  /* Specify that FFTW should expend a greater effort finding an
+     optimized plan. */
+  extern const FFTW_PATIENT : c_uint;
+
+  /* Specify that FFTW should expend an even greater effort finding an
+     optimized plan. */
+  extern const FFTW_EXHAUSTIVE : c_uint;
+
+  /* This is a special planning mode that is useful for querying
+     whether wisdom is available.  When using it, the plan is only
+     created when wisdom is available for the given problem; otherwise
+     a `null plan` is returned.  This can be combined with other flags
+     to create a plan if the wisdom available was created in that mode
+     (e.g., ``FFTW_WISDOM_ONLY | FFW_PATIENT``).  For more details on
+     this flag and the previous four, refer to `Section 4.3.2
+     <http://www.fftw.org/doc/Planner-Flags.html>`_ of the FFTW manual
+  */
+  extern const FFTW_WISDOM_ONLY : c_uint;
+
+
+  // Algorithm-restriction flags
+
+  /* Specify that an out-of-place transform is permitted to overwrite
+     its input array with arbitrary data.  This permits more efficient
+     algorithms to be used in some cases. */
+  // TODO: When we're ready to mention defaults, add: "This is the default for
+  // :proc:`plan_dft_c2r`. // NOTE: ...and hc2r once supported...
+  extern const FFTW_DESTROY_INPUT : c_uint;
+
+  /* Specify that an out-of-place transform cannot change its input
+     array. */
+  // TODO: When we're ready to mention defaults, add: This is the
+  // default for :proc:`plan_dft` and :proc:`plan_dft_r2c`. */
+  extern const FFTW_PRESERVE_INPUT : c_uint;
+
+  /* Specify that the algorithm may not impose any unusual alignment
+     requirements on the input/output arrays.  This flag should not be
+     necessary for current Chapel use since the planner will
+     automatically detect such cases.  For more details on this flag
+     and the previous two, refer to `Section 4.3.2
+     <http://www.fftw.org/doc/Planner-Flags.html>`_ of the FFTW manual.
+  */  // NOTE: But it will be if/when the new-array execute interface is supported
+  extern const FFTW_UNALIGNED : c_uint;
+
+
+  //
+  // This is a helper module to give us a "private" notion.  Once we
+  // have a private keyword, these could be hoisted to module scope
+  // and labeled private.
+  //
+  pragma "no doc"
+  module Private_FFTW {
+    //
+    // Check that two domain dimensions match, print an error if they
+    // don't, and return whether or not an error occurred.  Note that
+    // the 'inplace' argument is used to customize the error message
+    // for in-place and out-of-place cases.
+    //
+    proc checkDimMatch(inDom, outDom, dim, fnname, inplace=false) {
```

Comment:
It seems niggly, and I am not proposing changing this at this point, but it seems a little funky that 
"checkDimMatch" returns true if they do not match.  I do recognize that the documentation, and call site,
indicate that "true" means error so there's no functional error here.